include/veriblock/blockchain/blocktree.hpp

```diff
@@ -157,45 +153,11 @@ struct BlockTree {
     return current;
   }
 
-  bool load(ValidationState& state, bool bootstrapGenesis) {
-    // at this point we should have bootstrap block(s) in our storage
-    auto cursor = repo_->newCursor();
-
-    // load blocks into memory
-    std::vector<std::pair<height_t, std::unique_ptr<index_t>>> blocks;
-    for (cursor->seekToFirst(); cursor->isValid(); cursor->next()) {
-      auto value = cursor->value();
-
-      if (!checkProofOfWork(value.header, *param_)) {
-        return state.Error("load-bad-proof-of-work");
-      }
-
-      auto index = std::unique_ptr<index_t>(new index_t{});
-      *index = value;
-      blocks.push_back({value.height, std::move(index)});
-    }
-
-    size_t processedBlocks = 0;
-    size_t numBlocksForBootstrap = param_->numBlocksForBootstrap();
-    assert(block_index_.size() == 1);
-    std::sort(blocks.begin(), blocks.end());
-    for (const auto& item : blocks) {
-      index_t* index = item.second.get();
-      bool validateBlock =
-          bootstrapGenesis || processedBlocks++ > numBlocksForBootstrap;
-      if (acceptBlock(index->header, state, validateBlock)) {
-        return state.addStackFunction("load()");
-      }
-    }
-
-    return true;
-  }
-
   bool acceptBlock(const block_t& block,
                    ValidationState& state,
-                   bool validateBlock) {
+                   bool checkDifficulty) {
```

Comment:
This is not checkDifficulty anymore. It is checkDifficulty for BTC and checkByPrevious for VBK. Advised to rename to something more generic or split to two parameters.

Reply:
Fixed